### PR TITLE
Add `getInterest` rpc

### PR DESCRIPTION
**Pull request by canonbrother**
#### What kind of PR is this?:

/kind feature

#### Which issue(s) does this PR fixes?:

Fixes part of #488

### CODE REVIEW

**Comment by jingyi2811** [comment]
You need to change md file too

File: packages/jellyfish-api-core/src/category/loan.ts [hunk 1]
```diff
@@ -170,6 +170,17 @@ export class Loan {
     return await this.client.call('updateloantoken', [oldToken, newTokenDetails, utxos], 'number')
   }
 
+  /**
+   * Get interest info
+   *
+   * @param {string} id
+   * @param {string} token
```
Comment (jingyi2811) [inline packages/jellyfish-api-core/src/category/loan.ts:177]:
```suggestion
   * @param {string} [token]
```

File: packages/jellyfish-api-core/__tests__/category/loan/getInterest.test.ts [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+
+describe('Loan', () => {
+  const container = new LoanMasterNodeRegTestContainer()
+  const testing = Testing.create(container)
+  let interestTSLABlockHeight: number
+  let txUber: string
+
+  async function setup (): Promise<void> {
+    // token setup
+    const collateralAddress = await testing.container.getNewAddress()
```
Comment (jingyi2811) [inline packages/jellyfish-api-core/__tests__/category/loan/getInterest.test.ts:14]:
```suggestion
    const collateralAddress = await testing. generateAddress()
```
Like what you wrote in line 19

File: packages/jellyfish-api-core/__tests__/category/loan/getInterest.test.ts [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    await testing.generate(1)
+
+    const vaultAddress = await testing.generateAddress()
+    const vaultId = await testing.rpc.loan.createVault({
+      ownerAddress: vaultAddress,
+      loanSchemeId: 'scheme'
+    })
+    await testing.generate(1)
+
+    await testing.rpc.loan.createVault({
```
Comment (jingyi2811) [inline packages/jellyfish-api-core/__tests__/category/loan/getInterest.test.ts:81]: I think this createVault RPC call can be removed

**Comment by jingyi2811** [comment]
This PR is good to be approved. Just two minor things.

**Review by jingyi2811 — DISMISSED** [review]

File: packages/jellyfish-api-core/__tests__/category/loan/getInterest.test.ts [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+    // creationTx
+    {
+      const interests = await testing.rpc.loan.getInterest('scheme', txUber)
+      expect(interests.length).toStrictEqual(1)
+      expect(interests[0].token).toStrictEqual('UBER')
+    }
+
+    // id
+    {
+      const interests = await testing.rpc.loan.getInterest('scheme', '3')
```
Comment (surangap) [inline packages/jellyfish-api-core/__tests__/category/loan/getInterest.test.ts:162]: would be better if we can call RPC to get the id

File: packages/jellyfish-api-core/src/category/loan.ts [hunk 5]
```diff
@@ -170,6 +170,17 @@ export class Loan {
     return await this.client.call('updateloantoken', [oldToken, newTokenDetails, utxos], 'number')
   }
 
+  /**
+   * Get interest info
+   *
+   * @param {string} id
+   * @param {string} [token]
```
Comment (surangap) [inline packages/jellyfish-api-core/src/category/loan.ts:177]: can add descriptions about these params

File: packages/jellyfish-api-core/__tests__/category/loan/getInterest.test.ts (hunk 4 above)
Comment (canonbrother) [reply packages/jellyfish-api-core/__tests__/category/loan/getInterest.test.ts:162]: actually this style is more on readability wise
the point is not needed to know `3` is `AMZN` token but rather to tell this rpc support tokenId filter

**Review by jingyi2811 — APPROVED** [review]

**Review by surangap — APPROVED** [review]

**Review by fuxingloh — APPROVED** [review]

**Pull request merged**